From 4bb90cc9c883449ac9182164db8f038280c1131d Mon Sep 17 00:00:00 2001
From: Takuma Wood-Cooper <Takuma.Wood-Cooper@outlook.com>
Date: Tue, 20 Jun 2017 16:44:26 +0930
Subject: [PATCH 10/17] Made the frmTerm form Added frmAddModifyTerm button
 functionality Added frmTerm button functionality

Work done by Takuma Wood-Cooper
---
 .../VendorMaintenance/frmAddModifyTerm.cs          |   6 +-
 .../VendorMaintenance/frmTerm.Designer.cs          | 167 +++++++++++++++++++++
 Student Files/VendorMaintenance/frmTerm.cs         |  87 +++++++++++
 Student Files/VendorMaintenance/frmTerm.resx       | 120 +++++++++++++++
 4 files changed, 375 insertions(+), 5 deletions(-)
 create mode 100644 Student Files/VendorMaintenance/frmTerm.Designer.cs
 create mode 100644 Student Files/VendorMaintenance/frmTerm.cs
 create mode 100644 Student Files/VendorMaintenance/frmTerm.resx

diff --git a/Student Files/VendorMaintenance/frmAddModifyTerm.cs b/Student Files/VendorMaintenance/frmAddModifyTerm.cs
index dae6d30..e447439 100644
--- a/Student Files/VendorMaintenance/frmAddModifyTerm.cs	
+++ b/Student Files/VendorMaintenance/frmAddModifyTerm.cs	
@@ -18,17 +18,13 @@ namespace VendorMaintenance
         }
 
         public Term term;
+        public bool addTerm;
 
         private void frmAddModifyTerm_load(object sender, EventArgs e)
         {
 
         }
 
-        private void btnSave_Click(object sender, EventArgs e)
-        {
-
-        }
-
         private void btnAccept_Click(object sender, EventArgs e)
         {
             term = new Term();
diff --git a/Student Files/VendorMaintenance/frmTerm.Designer.cs b/Student Files/VendorMaintenance/frmTerm.Designer.cs
new file mode 100644
index 0000000..9a87f9c
--- /dev/null
+++ b/Student Files/VendorMaintenance/frmTerm.Designer.cs	
@@ -0,0 +1,167 @@
+﻿namespace VendorMaintenance
+{
+    partial class frmTerm
+    {
+        /// <summary>
+        /// Required designer variable.
+        /// </summary>
+        private System.ComponentModel.IContainer components = null;
+
+        /// <summary>
+        /// Clean up any resources being used.
+        /// </summary>
+        /// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing && (components != null))
+            {
+                components.Dispose();
+            }
+            base.Dispose(disposing);
+        }
+
+        #region Windows Form Designer generated code
+
+        /// <summary>
+        /// Required method for Designer support - do not modify
+        /// the contents of this method with the code editor.
+        /// </summary>
+        private void InitializeComponent()
+        {
+            this.label1 = new System.Windows.Forms.Label();
+            this.txtTermID = new System.Windows.Forms.TextBox();
+            this.label2 = new System.Windows.Forms.Label();
+            this.txtTermDesc = new System.Windows.Forms.TextBox();
+            this.label3 = new System.Windows.Forms.Label();
+            this.txtDueDays = new System.Windows.Forms.TextBox();
+            this.btnGetTerm = new System.Windows.Forms.Button();
+            this.btnAdd = new System.Windows.Forms.Button();
+            this.btnModify = new System.Windows.Forms.Button();
+            this.btnClear = new System.Windows.Forms.Button();
+            this.SuspendLayout();
+            // 
+            // label1
+            // 
+            this.label1.AutoSize = true;
+            this.label1.Location = new System.Drawing.Point(12, 9);
+            this.label1.Name = "label1";
+            this.label1.Size = new System.Drawing.Size(45, 13);
+            this.label1.TabIndex = 0;
+            this.label1.Text = "Term ID";
+            // 
+            // txtTermID
+            // 
+            this.txtTermID.Location = new System.Drawing.Point(15, 26);
+            this.txtTermID.Name = "txtTermID";
+            this.txtTermID.Size = new System.Drawing.Size(100, 20);
+            this.txtTermID.TabIndex = 1;
+            // 
+            // label2
+            // 
+            this.label2.AutoSize = true;
+            this.label2.Location = new System.Drawing.Point(13, 53);
+            this.label2.Name = "label2";
+            this.label2.Size = new System.Drawing.Size(87, 13);
+            this.label2.TabIndex = 2;
+            this.label2.Text = "Term Description";
+            // 
+            // txtTermDesc
+            // 
+            this.txtTermDesc.Location = new System.Drawing.Point(12, 70);
+            this.txtTermDesc.Multiline = true;
+            this.txtTermDesc.Name = "txtTermDesc";
+            this.txtTermDesc.Size = new System.Drawing.Size(260, 100);
+            this.txtTermDesc.TabIndex = 3;
+            // 
+            // label3
+            // 
+            this.label3.AutoSize = true;
+            this.label3.Location = new System.Drawing.Point(12, 177);
+            this.label3.Name = "label3";
+            this.label3.Size = new System.Drawing.Size(54, 13);
+            this.label3.TabIndex = 4;
+            this.label3.Text = "Due Days";
+            // 
+            // txtDueDays
+            // 
+            this.txtDueDays.Location = new System.Drawing.Point(12, 193);
+            this.txtDueDays.Name = "txtDueDays";
+            this.txtDueDays.Size = new System.Drawing.Size(260, 20);
+            this.txtDueDays.TabIndex = 5;
+            // 
+            // btnGetTerm
+            // 
+            this.btnGetTerm.Location = new System.Drawing.Point(121, 26);
+            this.btnGetTerm.Name = "btnGetTerm";
+            this.btnGetTerm.Size = new System.Drawing.Size(151, 23);
+            this.btnGetTerm.TabIndex = 6;
+            this.btnGetTerm.Text = "Get Term";
+            this.btnGetTerm.UseVisualStyleBackColor = true;
+            this.btnGetTerm.Click += new System.EventHandler(this.btnGetTerm_Click);
+            // 
+            // btnAdd
+            // 
+            this.btnAdd.Location = new System.Drawing.Point(12, 219);
+            this.btnAdd.Name = "btnAdd";
+            this.btnAdd.Size = new System.Drawing.Size(75, 23);
+            this.btnAdd.TabIndex = 7;
+            this.btnAdd.Text = "Add";
+            this.btnAdd.UseVisualStyleBackColor = true;
+            this.btnAdd.Click += new System.EventHandler(this.btnAdd_Click);
+            // 
+            // btnModify
+            // 
+            this.btnModify.Location = new System.Drawing.Point(94, 219);
+            this.btnModify.Name = "btnModify";
+            this.btnModify.Size = new System.Drawing.Size(75, 23);
+            this.btnModify.TabIndex = 8;
+            this.btnModify.Text = "Modify";
+            this.btnModify.UseVisualStyleBackColor = true;
+            this.btnModify.Click += new System.EventHandler(this.btnModify_Click);
+            // 
+            // btnClear
+            // 
+            this.btnClear.Location = new System.Drawing.Point(176, 219);
+            this.btnClear.Name = "btnClear";
+            this.btnClear.Size = new System.Drawing.Size(75, 23);
+            this.btnClear.TabIndex = 9;
+            this.btnClear.Text = "Clear";
+            this.btnClear.UseVisualStyleBackColor = true;
+            this.btnClear.Click += new System.EventHandler(this.btnClear_Click);
+            // 
+            // frmTerm
+            // 
+            this.AutoScaleDimensions = new System.Drawing.SizeF(6F, 13F);
+            this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
+            this.ClientSize = new System.Drawing.Size(284, 261);
+            this.Controls.Add(this.btnClear);
+            this.Controls.Add(this.btnModify);
+            this.Controls.Add(this.btnAdd);
+            this.Controls.Add(this.btnGetTerm);
+            this.Controls.Add(this.txtDueDays);
+            this.Controls.Add(this.label3);
+            this.Controls.Add(this.txtTermDesc);
+            this.Controls.Add(this.label2);
+            this.Controls.Add(this.txtTermID);
+            this.Controls.Add(this.label1);
+            this.Name = "frmTerm";
+            this.Text = "Terms";
+            this.ResumeLayout(false);
+            this.PerformLayout();
+
+        }
+
+        #endregion
+
+        private System.Windows.Forms.Label label1;
+        private System.Windows.Forms.TextBox txtTermID;
+        private System.Windows.Forms.Label label2;
+        private System.Windows.Forms.TextBox txtTermDesc;
+        private System.Windows.Forms.Label label3;
+        private System.Windows.Forms.TextBox txtDueDays;
+        private System.Windows.Forms.Button btnGetTerm;
+        private System.Windows.Forms.Button btnAdd;
+        private System.Windows.Forms.Button btnModify;
+        private System.Windows.Forms.Button btnClear;
+    }
+}
\ No newline at end of file
diff --git a/Student Files/VendorMaintenance/frmTerm.cs b/Student Files/VendorMaintenance/frmTerm.cs
new file mode 100644
index 0000000..fbff983
--- /dev/null
+++ b/Student Files/VendorMaintenance/frmTerm.cs	
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace VendorMaintenance
+{
+    public partial class frmTerm : Form
+    {
+        public frmTerm()
+        {
+            InitializeComponent();
+        }
+
+        public Term selectedTerm;
+
+        private void btnGetTerm_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                selectedTerm =
+                    (from Terms in DataContext.payables.Terms
+                    where Terms.TermsID == Convert.ToInt32(txtTermID.Text)
+                    select Terms).Single();
+                this.DisplayTerm();
+            }
+            catch (InvalidOperationException)
+            {
+                MessageBox.Show("No Term found with this ID. " +
+                    "Please re-enter the ID.", "this Term Not Found");
+                this.ClearControls();
+                txtTermID.Focus();
+            }
+            catch (Exception playBall)
+            {
+                MessageBox.Show(playBall.Message, playBall.GetType().ToString());
+            }
+        }
+
+        private void ClearControls()
+        {
+            txtTermID.Text = "";
+            txtTermDesc.Text = "";
+            txtDueDays.Text = "";
+        }
+
+        private void DisplayTerm()
+        {
+            txtTermID.Text = selectedTerm.TermsID.ToString();
+            txtTermDesc.Text = selectedTerm.Description.ToString();
+            txtDueDays.Text = selectedTerm.DueDays.ToString();
+        }
+
+        private void btnClear_Click(object sender, EventArgs e)
+        {
+            this.ClearControls();
+        }
+
+        private void btnAdd_Click(object sender, EventArgs e)
+        {
+            frmAddModifyTerm f;
+            f = new frmAddModifyTerm();
+
+            f.addTerm = true;
+
+            DialogResult result = f.ShowDialog();
+            if (result == DialogResult.OK)
+            {
+                selectedTerm = f.term;
+                txtTermID.Text= selectedTerm.TermsID.ToString();
+                this.DisplayTerm();
+            }
+        }
+
+        private void btnModify_Click(object sender, EventArgs e)
+        {
+
+        }
+
+
+    }
+}
diff --git a/Student Files/VendorMaintenance/frmTerm.resx b/Student Files/VendorMaintenance/frmTerm.resx
new file mode 100644
index 0000000..1af7de1
--- /dev/null
+++ b/Student Files/VendorMaintenance/frmTerm.resx	
@@ -0,0 +1,120 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<root>
+  <!-- 
+    Microsoft ResX Schema 
+    
+    Version 2.0
+    
+    The primary goals of this format is to allow a simple XML format 
+    that is mostly human readable. The generation and parsing of the 
+    various data types are done through the TypeConverter classes 
+    associated with the data types.
+    
+    Example:
+    
+    ... ado.net/XML headers & schema ...
+    <resheader name="resmimetype">text/microsoft-resx</resheader>
+    <resheader name="version">2.0</resheader>
+    <resheader name="reader">System.Resources.ResXResourceReader, System.Windows.Forms, ...</resheader>
+    <resheader name="writer">System.Resources.ResXResourceWriter, System.Windows.Forms, ...</resheader>
+    <data name="Name1"><value>this is my long string</value><comment>this is a comment</comment></data>
+    <data name="Color1" type="System.Drawing.Color, System.Drawing">Blue</data>
+    <data name="Bitmap1" mimetype="application/x-microsoft.net.object.binary.base64">
+        <value>[base64 mime encoded serialized .NET Framework object]</value>
+    </data>
+    <data name="Icon1" type="System.Drawing.Icon, System.Drawing" mimetype="application/x-microsoft.net.object.bytearray.base64">
+        <value>[base64 mime encoded string representing a byte array form of the .NET Framework object]</value>
+        <comment>This is a comment</comment>
+    </data>
+                
+    There are any number of "resheader" rows that contain simple 
+    name/value pairs.
+    
+    Each data row contains a name, and value. The row also contains a 
+    type or mimetype. Type corresponds to a .NET class that support 
+    text/value conversion through the TypeConverter architecture. 
+    Classes that don't support this are serialized and stored with the 
+    mimetype set.
+    
+    The mimetype is used for serialized objects, and tells the 
+    ResXResourceReader how to depersist the object. This is currently not 
+    extensible. For a given mimetype the value must be set accordingly:
+    
+    Note - application/x-microsoft.net.object.binary.base64 is the format 
+    that the ResXResourceWriter will generate, however the reader can 
+    read any of the formats listed below.
+    
+    mimetype: application/x-microsoft.net.object.binary.base64
+    value   : The object must be serialized with 
+            : System.Runtime.Serialization.Formatters.Binary.BinaryFormatter
+            : and then encoded with base64 encoding.
+    
+    mimetype: application/x-microsoft.net.object.soap.base64
+    value   : The object must be serialized with 
+            : System.Runtime.Serialization.Formatters.Soap.SoapFormatter
+            : and then encoded with base64 encoding.
+
+    mimetype: application/x-microsoft.net.object.bytearray.base64
+    value   : The object must be serialized into a byte array 
+            : using a System.ComponentModel.TypeConverter
+            : and then encoded with base64 encoding.
+    -->
+  <xsd:schema id="root" xmlns="" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:msdata="urn:schemas-microsoft-com:xml-msdata">
+    <xsd:import namespace="http://www.w3.org/XML/1998/namespace" />
+    <xsd:element name="root" msdata:IsDataSet="true">
+      <xsd:complexType>
+        <xsd:choice maxOccurs="unbounded">
+          <xsd:element name="metadata">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" />
+              </xsd:sequence>
+              <xsd:attribute name="name" use="required" type="xsd:string" />
+              <xsd:attribute name="type" type="xsd:string" />
+              <xsd:attribute name="mimetype" type="xsd:string" />
+              <xsd:attribute ref="xml:space" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="assembly">
+            <xsd:complexType>
+              <xsd:attribute name="alias" type="xsd:string" />
+              <xsd:attribute name="name" type="xsd:string" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="data">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+                <xsd:element name="comment" type="xsd:string" minOccurs="0" msdata:Ordinal="2" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" msdata:Ordinal="1" />
+              <xsd:attribute name="type" type="xsd:string" msdata:Ordinal="3" />
+              <xsd:attribute name="mimetype" type="xsd:string" msdata:Ordinal="4" />
+              <xsd:attribute ref="xml:space" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="resheader">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" />
+            </xsd:complexType>
+          </xsd:element>
+        </xsd:choice>
+      </xsd:complexType>
+    </xsd:element>
+  </xsd:schema>
+  <resheader name="resmimetype">
+    <value>text/microsoft-resx</value>
+  </resheader>
+  <resheader name="version">
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+</root>
\ No newline at end of file
-- 
2.11.0.windows.1

